ee/clickhouse/queries/groups_join_query.py

```diff
@@ -29,10 +27,16 @@ def __init__(
         self._team_id = team_id
         self._column_optimizer = column_optimizer or ColumnOptimizer(self._filter, self._team_id)
 
-    def get_join_query(self) -> Tuple[str, Dict]:
+    def get_join_query(self, group_join_keys: Optional[List[str]] = None) -> Tuple[str, Dict]:
         join_queries, params = [], {}
 
-        for group_type_index in self._column_optimizer.group_types_to_query:
+        if group_join_keys:
+            if len(group_join_keys) != len(self._column_optimizer.group_types_to_query):
+                raise ValidationError("When specifying group_join_keys, number should be equal to groups to be queried")
```

Comment:
This error is not gonna say much to the user. Is this an internal error? Should we raise an `ValueError` instead?